File: project-in-dotnet/Vehicle_Configurator_Backend/Vehicle_Configurator/Domain/Entities/Component.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vehicle_Configurator.Domain.Entities
{
    [Table("component")]
    public class Component
    {
        [Key]
        [Column("comp_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int CompId { get; set; }

        [Column("comp_name")]
        public string CompName { get; set; }

        public ICollection<VehicleDetail> VehicleDetails { get; set; }
        public ICollection<InvoiceDetail> InvoiceDetails { get; set; }
        public ICollection<AlternateComponentMaster> DefaultComponents { get; set; }
        public ICollection<AlternateComponentMaster> AltComponents { get; set; }
    }
}

File: project-in-java/backend/Vehicle_Configurator-main/src/main/java/com/example/entity/Component.java
package com.example.entity;

import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import lombok.Data;


@Entity
@Data
public class Component {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private int comp_id;
	
	private String comp_name;
	
	@OneToMany(mappedBy = "component")
	@JsonIgnore
	private Set<Vehicle_detail> vehicleDetails;
	
	@OneToMany(mappedBy = "component")
	@JsonIgnore
	private Set<Invoice_detail> invoiceDetails;
	
	@OneToMany(cascade=CascadeType.ALL)
	@JoinColumn(name="comp_id", referencedColumnName="comp_id")
	private Set<Alternate_Component_Master> defaultComp;
	
	@OneToMany(cascade=CascadeType.ALL)
	@JoinColumn(name="alt_comp_id", referencedColumnName="comp_id")
	private Set<Alternate_Component_Master> altComp;
}
